### ISSUE TITLE
`read_sql_query` returns dataframe with duplicates in its index

### ISSUE THREAD

**Issue by zseder** [awslabs/aws-data-wrangler#521]
With version `2.3.0`, when running a query and fetching its result by using `read_sql_query(ctas_approach=True)` (which is default), the resulting dataframe will contain duplicated entries in its index. By passing `ctas_approach=False`, the resulting index will be unique.

I guess it has something to do with chunked reading and calling `pd.concat()`. I tried to dig into the source to find the bug with no success.

When using `ctas_approach=True, keep_files=True`, and then reading directly the raw files by `pd.read_parquet('s3://...')`, the resulting index will be unique.

**Comment by maxispeicher** [comment]
Can you tell a little bit more about the concrete setup respectively the kind of query you are running? I was just trying to reproduce it on my side with a simple `SELECT * FROM mytable` and I get a unique index for `ctas_approach=True` and `ctas_approach=False`.

**Comment by zseder** [comment]
Checking my version:
```
import awswrangler as wr
wr.__version__
```
```
2.3.0
```

Using `read_sql_query()` and printing output shapes
```
df = wr.athena.read_sql_query('SELECT * FROM db.table', database=db, ctas_approach=True)
print(df.index.shape[0], df.index.unique().shape[0])
```

```
15264 14240
```
Here we can see the difference.

The type of index is a regular `Int64Index` this way. But when I read with `ctas_approach=False`, I get a `RangeIndex` and that is of course unique.
```
df.index
Out[7]:
Int64Index([   0,    1,    2,    3,    4,    5,    6,    7,    8,    9,
            ...
            1014, 1015, 1016, 1017, 1018, 1019, 1020, 1021, 1022, 1023],
           dtype='int64', length=15264)
```

The strange thing is that I get this behavior even if the table is backed by a single-row-group parquet file Which I guess cannot even be read parallel (I'm just guessing here).
The similar thing happens with multifile parquet dataset as well.

**Comment by maxispeicher** [comment]
I'm not sure where the underlying problem is respectively how to fix it, but I think I can give you a small update when this happens. If you have query with unique results also the index will be unique. However if you have duplicated rows in your query also the index seems to be duplicated:
```python
import awswrangler as wr

df = wr.athena.read_sql_query(
  sql="SELECT * FROM mydatabase.my_unique_table", 
  database="mydatabase", 
  ctas_approach=True
)
print(df.index.shape[0])  # Will output 4996
print(df.index.unique().shape[0])  # Will output 4996

df = wr.athena.read_sql_query(
  sql="SELECT * FROM mydatabase.my_unique_table UNION ALL SELECT * FROM mydatabase.my_unique_table", 
  database="mydatabase", 
  ctas_approach=True
)
print(df.index.shape[0])  # Will output 9992
print(df.index.unique().shape[0])  # Will output 4996
```

**Comment by zseder** [comment]
At this particular table that I have, all values are unique, both the following queries give the same result:
```
SELECT COUNT() FROM db.table
SELECT COUNT() FROM (SELECT DISTINCT * FROM db.table)
```
(and the result is the same as the larger number in https://github.com/awslabs/aws-data-wrangler/issues/521#issuecomment-766667831, `15264`)

**Comment by maxispeicher** [comment]
Hm ok this is strange. Just to confirm, when you use `keep_files=True` is there only one parquet file in the result location or are there multiple?

**Comment by maxispeicher** [comment]
I've tried to fix this on a new branch. Could you try installing it from my branch and see if it fixes your issue. If that's the case, I would polish the solution:
```
pip uninstall awswrangler -y
pip install git+https://github.com/maxispeicher/aws-data-wrangler.git@athena_ctas_unique_index
```

**Comment by zseder** [comment]
> Hm ok this is strange. Just to confirm, when you use `keep_files=True` is there only one parquet file in the result location or are there multiple?

There are two files.

**Comment by zseder** [comment]
> I've tried to fix this on a new branch. Could you try installing it from my branch and see if it fixes your issue. If that's the case, I would polish the solution:
> 
> ```
> pip uninstall awswrangler -y
> pip install git+https://github.com/maxispeicher/aws-data-wrangler.git@athena_ctas_unique_index
> ```

Yes, this branch solves this issue, the resulting index is a (unique) `RangeIndex`

**Comment by maxispeicher** [comment]
> > Hm ok this is strange. Just to confirm, when you use `keep_files=True` is there only one parquet file in the result location or are there multiple?
> 
> There are two files.

Ok then it makes sense again. I was just assuming that it was one file as `pd.read_parquet('s3://...')` only takes a single argument and I forgot that it can also be a directory (IIRC).

But happy to hear that the branch solves the issue :)

**Comment by igorborgest** [comment]
Released on version [2.4.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.4.0).
